### ISSUE TITLE
libhwy.so.1.0.0: undefined reference to `__atomic_store_8'

### ISSUE THREAD

**Issue by malaterre** [google/highway#887]
It seems something has changed recently (or is it gcc-12). But some arches are now failing to build highway. Typical error message is now:

```
 && /usr/bin/c++ -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -DHWY_BROKEN_EMU128=1 -Wno-psabi -Wdate-time -D_FORTIFY_SOURCE=2 -Wl,-z,relro -Wl,-z,now     -fPIE -pie CMakeFiles/hwy_benchmark.dir/hwy/examples/benchmark.cc.o -o examples/hwy_benchmark  -Wl,-rpath,/<<PKGBUILDDIR>>/obj-arm-linux-gnueabi  libhwy.so.1.0.0 && :
/usr/bin/ld: libhwy.so.1.0.0: undefined reference to `__atomic_store_8'
collect2: error: ld returned 1 exit status
```

* https://buildd.debian.org/status/fetch.php?pkg=highway&arch=armel&ver=1.0.0-1&stamp=1659387322&raw=0

**Comment by malaterre** [comment]
Possible patch would be something like this:

* https://github.com/libjxl/libjxl/commit/87fe7c16e1fb2e21b6a1dca26782950ae1559d99

**Comment by jan-wassenberg** [comment]
hm, would it be sufficient to add -latomic?

**Comment by malaterre** [comment]
> hm, would it be sufficient to add -latomic?

This will fail for msvc. But yes unconditionally adding -latomic is just fine.

**Comment by malaterre** [comment]
Testing currently:

* https://salsa.debian.org/debian-phototools-team/highway/-/blob/debian/1.0.0-3/debian/patches/atomic.patch

**Comment by jan-wassenberg** [comment]
Sounds good, sending a similar change, thanks :)

**Comment by novomesk** [comment]
I had similar problem on `ppc`:

```
[36/38] : && /usr/bin/powerpc-unknown-linux-gnu-g++ -Og -mcpu=7400 -Wl,-O1 -Wl,--as-needed     -fPIE -pie CMakeFiles/hwy_list_targets.dir/hwy/tests/list_targets.cc.o -o hwy_list_targets  -Wl,-rpath,/var/tmp/portage/dev-cpp/highway-1.0.0/work/highway-1.0.0_build-.ppc  libhwy.so.1.0.0 && cd /var/tmp/portage/dev-cpp/highway-1.0.0/work/highway-1.0.0_build-.ppc && /var/tmp/portage/dev-cpp/highway-1.0.0/work/highway-1.0.0_build-.ppc/hwy_list_targets || ( exit 0 )
/usr/lib/gcc/powerpc-unknown-linux-gnu/11.2.1/../../../../powerpc-unknown-linux-gnu/bin/ld: libhwy.so.1.0.0: undefined reference to `__atomic_store_8'
collect2: error: ld returned 1 exit status
[37/38] /usr/bin/powerpc-unknown-linux-gnu-g++ -DHWY_SHARED_DEFINE -I/var/tmp/portage/dev-cpp/highway-1.0.0/work/highway-1.0.0  -Og -mcpu=7400 -fPIE -fvisibility=hidden -fvisibility-inlines-hidden -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\" -fmerge-all-constants -Wall -Wextra -Wconversion -Wsign-conversion -Wvla -Wnon-virtual-dtor -fmath-errno -fno-exceptions -MD -MT CMakeFiles/hwy_benchmark.dir/hwy/examples/benchmark.cc.o -MF CMakeFiles/hwy_benchmark.dir/hwy/examples/benchmark.cc.o.d -o CMakeFiles/hwy_benchmark.dir/hwy/examples/benchmark.cc.o -c /var/tmp/portage/dev-cpp/highway-1.0.0/work/highway-1.0.0/hwy/examples/benchmark.cc
[38/38] : && /usr/bin/powerpc-unknown-linux-gnu-g++ -Og -mcpu=7400 -Wl,-O1 -Wl,--as-needed     -fPIE -pie CMakeFiles/hwy_benchmark.dir/hwy/examples/benchmark.cc.o -o examples/hwy_benchmark  -Wl,-rpath,/var/tmp/portage/dev-cpp/highway-1.0.0/work/highway-1.0.0_build-.ppc  libhwy.so.1.0.0 && :
FAILED: examples/hwy_benchmark 
: && /usr/bin/powerpc-unknown-linux-gnu-g++ -Og -mcpu=7400 -Wl,-O1 -Wl,--as-needed     -fPIE -pie CMakeFiles/hwy_benchmark.dir/hwy/examples/benchmark.cc.o -o examples/hwy_benchmark  -Wl,-rpath,/var/tmp/portage/dev-cpp/highway-1.0.0/work/highway-1.0.0_build-.ppc  libhwy.so.1.0.0 && :
/usr/lib/gcc/powerpc-unknown-linux-gnu/11.2.1/../../../../powerpc-unknown-linux-gnu/bin/ld: libhwy.so.1.0.0: undefined reference to `__atomic_store_8'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
 * ERROR: dev-cpp/highway-1.0.0::gentoo failed (compile phase):
```

**Comment by jan-wassenberg** [comment]
Hi @novomesk , is this despite the above fix? For your system, does the `UNIX AND NOT APPLE` CMake check evaluate to true?

**Comment by novomesk** [comment]
It happened only in released 1.0.0

When using snapshot from git, it worked.

**Comment by jan-wassenberg** [comment]
Thank you for confirming. There will be a 1.0.1 release soon, probably within a few days.
